### PR TITLE
Feat: Add ProgressIndicator Component

### PR DESCRIPTION
**Pull request by obrucheoghene**
## Description

Create the ProgressIndicator component #676

Screenshot 2024-12-21 at 9 59 27 PM

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?
Implement different stories  for the storybook

## Checklist:

- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings

### CODE REVIEW

File: mods/webui/src/stories/progessindicator/ProgressIndicator.styles.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { styled } from "@mui/material/styles";
+
+export const StyledProgressContainer = styled("div")(({ theme }) => ({
```
Comment (psanders) [inline mods/webui/src/stories/progessindicator/ProgressIndicator.styles.ts]: The theme is unused.

File: mods/webui/src/stories/progessindicator/ProgressIndicator.styles.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  height: "4px",
+  backgroundColor: theme.palette.primary[200]
+}));
+
+export const StyledProgressIndicator = styled("div")(({ theme }) => ({
+  height: "100%",
+  backgroundColor: theme.palette.primary[500]
+}));
+
+export const StyledIndicatorLabelList = styled("ol")(({ theme }) => ({
```
Comment (psanders) [inline mods/webui/src/stories/progessindicator/ProgressIndicator.styles.ts]: The theme is unused.

File: mods/webui/src/stories/progessindicator/ProgressIndicator.styles.ts [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+  display: "flex",
+  flexDirection: "row",
+  justifyContent: "space-between",
+  listStyle: "none",
+  padding: "0px",
+  margin: "0px",
+  gap: "10px"
+}));
+
+export const StyledIndicatorLabel = styled("li")(({ theme }) => ({}));
```
Comment (psanders) [inline mods/webui/src/stories/progessindicator/ProgressIndicator.styles.ts]: The theme is unused.

File: mods/webui/src/stories/progessindicator/ProgressIndicator.tsx [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+          sx={{
+            width: `${progress}%`
+          }}
+        />
+      </StyledProgressBar>
+
+      <StyledIndicatorLabelList>
+        {steps.map((step, i) => (
+          <StyledIndicatorLabel key={i}>
+            <Typography variant="body-small">{step}</Typography>
```
Comment (psanders) [inline mods/webui/src/stories/progessindicator/ProgressIndicator.tsx:49]: Can you take another pass at the styles? The remaining steps appear unhighlighted (for lack of a better term).

Screenshot 2024-12-21 at 5 22 13 PM

**Review by psanders — CHANGES_REQUESTED** [review]
This looks super smooth! I added a comment to review the styles.

**Comment by obrucheoghene** [comment]
Unused theme removed.
Thank you

File: mods/webui/src/stories/progessindicator/ProgressIndicator.tsx (hunk 4 above)
Comment (psanders) [reply mods/webui/src/stories/progessindicator/ProgressIndicator.tsx:49]: @obrucheoghene, this portion still needs addressing.

Comment (obrucheoghene) [reply mods/webui/src/stories/progessindicator/ProgressIndicator.tsx:49]: fixed.

**Comment by psanders** [comment]
@obrucheoghene 

All set here! Fantastic work, my friend.

**Comment by obrucheoghene** [comment]
@psanders 
Thanks.

**Pull request merged**